### PR TITLE
Add initial support for auto-instrumentation

### PR DESCRIPTION
**Pull request by pavolloffay**
Updates #455 

This PR adds functionality to inject OpenTelemetry auto-instrumentation (at this point only javaagent) into a pod via admission controller. Note that this is the smallest PR I could submit to support at least something end-to-end. There will be follow up PRs to provide additional features.

Notable changes 
* new CR to define SDK and instrumentation configuration
* javaagent is injected and configured if `java.inst.opentelemetry.io/inject: "true"` is specified. The value van be the instrumentation CR name.
* the javaagent injection uses init container to copy javaagent jar from init container image to a volume that is mounted to the application container
* the configuration of the instrumentation and SDK is done via environment variables

#### CR
```yaml
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryInst
metadata:
  name: java-instrumentation
spec:
  exporter:
    endpoint: http://otel-collector:1
  java:
    image: ghcr.io/pavolloffay/otel-javaagent:1.5.3
```

#### Follow up PRs
* Support better service name
* Inject k8s resources - namespace, cluster etc.
* add sampling configuration to CR
* add support for other languages
* add status .e.g. show which workloads are instrumented and what is the sampling configuration...
* add flag to set default javaagent image
* add e2e test

### CODE REVIEW

**Comment by anuraaga** [comment]
@jpkrohling Just curious if this is ready to get in, with more followups for e.g. e2e testing. Looking forward to extending to add more languages :)

**Comment by jpkrohling** [comment]
I'll try to get a final review later today. If @VineethReddy02 can do it before me and approve, I'll merge it without my final review.

**Review by anuraaga — APPROVED** [review]

File: README.md
Comment (jpkrohling) [inline README.md:154]: We do not have forced line wraps in this file, as editors can do it as well. While I don't have a particular preference, I do prefer the same style to be used consistently.

File: README.md
Comment (jpkrohling) [inline README.md:170]:
```suggestion
In addition to the annotation, the following `CR` has to be created. The `Instrumentation` resource
```

File: README.md
Comment (jpkrohling) [inline README.md:167]: What happens in this case? If it follows the same resolution mechanism as the sidecar injection, it might make sense to mention here.

File: README.md
Comment (jpkrohling) [inline README.md:188]:
```suggestion
   The image must contain the Java agent JAR at `/javaagent.jar`, and the operator will copy the JAR to a shared volume mounted to the application pod.
```

@jstickler would be proud of me, reducing the usage of the passive voice :)

**Review by jpkrohling — APPROVED** [review]
LGTM. Some minor changes to the readme, but nothing serious. I have asked @owais to provide short feedback on the effects of this PR in downstream forks and will merge this later today based on his feedback.

Feel free to work on the readme changes in a follow-up PR.

**Comment by jpkrohling** [comment]
@owais just confirmed that this is great from his side as well. Merging!

**Pull request merged**
